tests/sparseml/modifiers/pruning/constant/test_pytorch.py

```diff
@@ -0,0 +1,173 @@
+def test_constant_pruning_modifier_e2e(model, optimizer):
+    # move imports inside so that pytorch is not required unless
+    # running this test
+
+    from sparseml.core import State
+    from sparseml.core.event import Event, EventType
+    from sparseml.core.framework import Framework
+    from sparseml.modifiers.pruning.constant.pytorch import (
+        ConstantPruningModifierPyTorch,
+    )
+    from sparseml.pytorch.utils import tensor_sparsity
+
+    expected_sparsities = {
+        name: tensor_sparsity(param.data)
+        for name, param in model.named_parameters()
+        if "weight" in name
+    }
+
+    # init modifier with model
+
+    state = State(framework=Framework.pytorch)
+    state.update(
+        model=model,
+        optimizer=optimizer(model.parameters(), lr=0.1),
+        start=0,
+    )
+    modifier = ConstantPruningModifierPyTorch(
+        targets="__ALL_PRUNABLE__",
+        start=0,
+        end=1,
+        update=0.5,
+    )
+    modifier.initialize(state)
+
+    # mess up model sparsity
+
+    model = _make_dense(model)
+    manipulated_sparsities = {
+        name: tensor_sparsity(param.data)
+        for name, param in model.named_parameters()
+        if "weight" in name
+    }
+    assert manipulated_sparsities != expected_sparsities, "Sparsity manipulation failed"
+
+    # apply modifier
+
+    modifier.on_update(state, event=Event(type_=EventType.OPTIM_PRE_STEP))
+    modifier.on_update(state, event=Event(type_=EventType.OPTIM_POST_STEP))
+    modifier.finalize(state)
+    modifier.on_end(state, None)
+
+    # sparsity should restored by ConstantPruningModifierPyTorch
+
+    actual_sparsities = {
+        name: tensor_sparsity(param.data)
+        for name, param in model.named_parameters()
+        if "weight" in name
+    }
+    assert actual_sparsities == expected_sparsities, "Sparsity was not constant"
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+def test_constant_pruning_pytorch_is_registered():
+    from sparseml.core.factory import ModifierFactory
+    from sparseml.core.framework import Framework
+    from sparseml.modifiers.pruning.constant.pytorch import (
+        ConstantPruningModifierPyTorch,
+    )
+
+    kwargs = dict(
+        start_epoch=5.0,
+        end_epoch=15.0,
+        targets="__ALL_PRUNABLE__",
+    )
+    _setup_modifier_factory()
+    type_ = ModifierFactory.create(
+        type_="ConstantPruningModifier",
+        framework=Framework.pytorch,
+        allow_experimental=False,
+        allow_registered=True,
+        **kwargs,
+    )
+
+    assert isinstance(
+        type_, ConstantPruningModifierPyTorch
+    ), "PyTorch ConstantPruningModifier not registered"
+
+
+def _setup_modifier_factory():
+    from sparseml.core.factory import ModifierFactory
+
+    ModifierFactory.refresh()
+    assert ModifierFactory._loaded, "ModifierFactory not loaded"
```

Comment:
This seems like it would be a useful helper function outside of this class, maybe we could move it to a general test helpers file

Reply:
Agree moving this to a helper!